### PR TITLE
add /api/articles/:article_id endpoint

### PR DESCRIPTION
**Pull request by robert-matthew-brooks**
added error handler for catching custom errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
...
+    test('response data should be on an "article" key with a 200 http status code', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveProperty('article');
+        });
+    });
+
+    test('each article should have correct object layout', () => {
```
Comment (RichardNeat) [inline __tests__/app.test.js]: We are only getting one object back on this endpoint - so we could be more accurate with the test description. Also as we are using a test database we know exactly what this data will look like so we could be more specific and hard code this expect response.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
...
+
+            supertestRequests.push(article);
+        }
+
+        return Promise.all(supertestRequests)
+        .then(articles => {
+            for (const article in articles) {
+                expect(articles[article]).toMatchObject(objectLayout);
+            }
+        });
```
Comment (RichardNeat) [inline __tests__/app.test.js]: Cool idea to check every item in the database, but possibly a bit of an over test. If it works for one and gives us the response we expect, then we can trust it to work with any id.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
...
+    });
+
+    test('response data should match the provided seed data', () => {
```
Comment (RichardNeat) [inline __tests__/app.test.js]: Similar comment here, we don't need to worry about manipulating the data this heavily when we could just hard code the response that we expect when making a request for the specific ID.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -77,4 +77,100 @@ describe('GET /api/topics', () => {
...
+
+    describe('error handling', () => {
+        it('should return a http 404 error if article not in database', () => {
+            return request(app)
+            .get('/api/articles/99')
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe('article not found');
+            });
+        });
```
Comment (RichardNeat) [inline __tests__/app.test.js:280]: Great to have a 404 test here, what if I made a request to /api/articles/banana? What error would we get here? Could we add a test for it?

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Hi Robert - a few changes in comments below please then resubmit :)

**Review by Slyose — APPROVED** [review]
Good response to feedback, merge away!

**Pull request merged**